packages/@okta/vuepress-theme-prose/components/OnThisPage.vue

```diff
@@ -66,21 +67,17 @@
           document.documentElement.scrollTop,
           document.body.scrollTop
         )
-        for (let i = 0; i < anchors.length; i++) {
-          const anchor = anchors[i]
-          const nextAnchor = anchors[i + 1]
 
-          const isActive = i === 0 && scrollTop === 0
-            || (scrollTop >= anchor.parentElement.offsetTop - document.querySelector('.fixed-header').clientHeight - 45
-              && (!nextAnchor || scrollTop < nextAnchor.parentElement.offsetTop - document.querySelector('.fixed-header').clientHeight - 45))
-          
-          
-          
-          if (isActive && decodeURIComponent(this.$route.hash) !== decodeURIComponent(anchor.hash)) {
-            this.activeAnchor = anchor.hash;
-            
-            return
-          }
+        const paddedHeaderHeight = document.querySelector('.fixed-header').clientHeight + LAYOUT_CONSTANTS.HEADER_TO_CONTENT_GAP;
+        const anchorOffsets = anchors.map(anchor => anchor.parentElement.offsetTop);
+        const anchorOffsetPairs = anchorOffsets.map((anchorOffset, index, anchorOffsets) => [anchorOffset, anchorOffsets[index + 1]]);
```

Comment:
Maybe cache `anchors`, `paddedHeaderHeight`, `anchorOffsetPairs` to component variables like was done in `ContentPage`?
Update them on mount and after change of `this.activeAnchor`, instead of every scroll event.
Will improve performance.